sdk/search/Azure.Search.Documents/src/Indexes/Models/IndexingParametersExtensions.cs

```diff
@@ -0,0 +1,220 @@
+        public static IndexingParameters ParseJsonLines(this IndexingParameters parameters) =>
+            Configure(parameters, ParsingModeKey, "jsonLines");
+
+        /// <summary>
+        /// Tells the indexer to assume that all blobs contain JSON arrays, which it will then parse such that each JSON object in each array will
+        /// map to a single document in the search index.
+        /// See <see href="https://docs.microsoft.com/azure/search/search-howto-index-json-blobs" /> for details.
+        /// </summary>
+        /// <param name="parameters">The <see cref="IndexingParameters"/> to configure.</param>
+        /// <param name="documentRoot">
+        /// An optional <see href="https://tools.ietf.org/html/rfc6901">JSON Pointer</see> that tells the indexer how to find the JSON array if it's not the top-level JSON property of each blob.
+        /// If this parameter is null or empty, the indexer will assume that the JSON array can be found in the top-level JSON property of each blob.
+        /// Default is null.
+        /// </param>
+        /// <remarks>
+        /// This option only applies to indexers that index Azure Blob Storage.
+        /// </remarks>
+        /// <returns>The <see cref="IndexingParameters"/> instance.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="parameters"/> is null.</exception>
+        public static IndexingParameters ParseJsonArrays(this IndexingParameters parameters, string documentRoot = null)
+        {
+            Configure(parameters, ParsingModeKey, "jsonArray");
+
+            if (!string.IsNullOrEmpty(documentRoot))
+            {
+                Configure(parameters, "documentRoot", documentRoot);
+            }
+
+            return parameters;
+        }
+
+        /// <summary>
+        /// Tells the indexer to assume that all blobs are delimited text files. Currently only comma-separated value (CSV) text files are supported.
+        /// See <see href="https://docs.microsoft.com/azure/search/search-howto-index-csv-blobs" /> for details.
+        /// </summary>
+        /// <param name="parameters">The <see cref="IndexingParameters"/> to configure.</param>
+        /// <param name="headers">
+        /// Specifies column headers that the indexer will use to map values to specific fields in the search index. If you don't specify any
+        /// headers, the indexer assumes that the first non-blank line of each blob contains comma-separated headers.
+        /// </param>
+        /// <remarks>
+        /// This option only applies to indexers that index Azure Blob Storage.
+        /// </remarks>
+        /// <returns>The <see cref="IndexingParameters"/> instance.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="parameters"/> is null.</exception>
+        public static IndexingParameters ParseDelimitedTextFiles(this IndexingParameters parameters, params string[] headers)
```

Comment:
Should we take a separator type or format (that's only CSV today)?  The params gives me a bit of caution in case it's going to be a string.

Reply:
That's the format that the service expects for this property in the bag. I'll defer to @bleroy and @jennifermarsman if it would be sensible to allow the user to define others for possible future use.

Reply:
In the portal, we have an open text box to accept a string for the delimiter character (it does default to comma).  So yes, I do think we need to add functionality to accept that here.  

There is also a boolean (checkbox in the portal) to specify whether the first line contains headers.  If it is unchecked, the headers open text box appears.  The below logic looks correct.